health-services/ExerciseSample/app/src/main/java/com/example/exercise/HealthServicesManager.kt

```diff
@@ -200,26 +188,33 @@ class HealthServicesManager @Inject constructor(
      * messages. When there are no more subscribers, or when the coroutine scope of [shareIn] is
      * cancelled, this flow will unregister the listener.
      *
-     * A shared flow is used because only a single [ExerciseUpdateListener] can be reigstered at a
+     * A shared flow is used because only a single [ExerciseUpdateListener] can be registered at a
      * time, even if there are multiple consumers of the flow.
      *
      * [callbackFlow] is used to bridge between a callback-based API and Kotlin flows.
      */
     @OptIn(ExperimentalCoroutinesApi::class)
-    val exerciseUpdateFlow = callbackFlow<ExerciseMessage> {
-        val listener = object : ExerciseUpdateListener {
-            override fun onExerciseUpdate(update: ExerciseUpdate) {
+    val exerciseUpdateFlow = callbackFlow {
+        val callback = object : ExerciseUpdateCallback {
+            override fun onExerciseUpdateReceived(update: ExerciseUpdate) {
                 coroutineScope.runCatching {
                     trySendBlocking(ExerciseMessage.ExerciseUpdateMessage(update))
                 }
             }
 
-            override fun onLapSummary(lapSummary: ExerciseLapSummary) {
+            override fun onLapSummaryReceived(lapSummary: ExerciseLapSummary) {
                 coroutineScope.runCatching {
                     trySendBlocking(ExerciseMessage.LapSummaryMessage(lapSummary))
                 }
             }
 
+            override fun onRegistered() {
```

Comment:
What do we recommend apps do in `onRegistered` and `onRegistrationFailed`? Is there anything sensible apps can do?